packages/quantic/force-app/main/default/lwc/quanticResultField/quanticResultField.js

```diff
@@ -0,0 +1,69 @@
+import {LightningElement, api} from 'lwc';
+// @ts-ignore
+import dateTemplate from './dateTemplate.html';
+// @ts-ignore
+import multiValueTemplate from './multiValueTemplate.html';
+// @ts-ignore
+import numberTemplate from './numberTemplate.html';
+// @ts-ignore
+import stringTemplate from './stringTemplate.html';
+
+/** @typedef {import("coveo").Result} Result */
+
+/**
+ * The `QuanticResultField` component properly displays a given result field according to its type.
+ * @category Result Template
+ * @example
+ * <template if:true={result.raw.source}>
+ *   <c-quantic-result-field result={result} label="Source" field="source" type="string"></c-quantic-result-field>
+ * </template>
+ */
+export default class QuanticResultField extends LightningElement {
+  /**
+   * The [result item](https://docs.coveo.com/en/headless/latest/reference/search/controllers/result-list/#result) to use.
+   * @api
+   * @type {Result}
+   */
+  @api result;
+  /**
+   * (Optional) The label to display.
+   * @api
+   * @type {string}
+   * @defaultValue `undefined`
+   */
+  @api label;
+  /**
+   * The field name whose values you want to display.
+   * @api
+   * @type {string}
+   */
+  @api field;
+  /**
+   * The field type whose values you want to display.
+   * @type { 'string' | 'date' | 'number' | 'multi'}
+   */
+  @api type;
+
+  templateTypeMap = {
+    multi: multiValueTemplate,
+    number: numberTemplate,
+    date: dateTemplate,
+    string: stringTemplate,
+    default: stringTemplate,
+  };
+
+  connectedCallback() {
+    const templateTypes = Object.keys(this.templateTypeMap).slice(0, 4);
+    if (!templateTypes.includes(this.type)) {
+      console.error(
```

Comment:
Just saying but if instead of the multiple template files we had one quanticResultField.html with some conditional rendering we could display the error in red like we do for many other components. 👀

Reply:
...or we could show an error template in such case. I like how the component uses specific templates, but I don't have strong opinions on that matter.

Reply:
I will do that! a new template for the error to get the best of both words!

Reply:
Just for my own knowledge, what's the advantages you like of the templates?

Reply:
In this situation it looks clean to me to have separate templates. I'd say it is more readable. Each template is super simple and its scope is well defined. It is very easy to know what is going to be rendered.

When everything is done in a single template, it is often more difficult to get the same understanding of the component.

For instance, if you have a single template, it is difficult for a developer to know whether an error message can be displayed at the same time as the value. With separate templates, it is obvious.

Reply:
I could easily imagine situations where using separate templates would not make sense. It is a matter of using the right tool for the job.

Reply:
That's valid. I think i have an aversion to files with 1 or 2 lines in them

Reply:
I'm fine with the templates so just to try to illustrate my personal feelings about it I'd compare it to putting each pair of my socks in a separate drawer. I can label the drawers and it's easy to find the socks I want but it's cumbersome. 😛

Reply:
I like the analogy :smile:. Honestly, I don't mind whether the component uses a single template or many.

Reply:
Neither do I really we can keep it as it is now. 👍